editor/src/clj/editor/grid.clj

```diff
@@ -13,192 +13,385 @@
-  [^AABB aabb grid-size]
-  (types/->AABB (Point3d. (grid-snap-down (-> aabb types/min-p .x) grid-size)
-                      (grid-snap-down (-> aabb types/min-p .y) grid-size)
-                      (grid-snap-down (-> aabb types/min-p .z) grid-size))
-            (Point3d. (grid-snap-up   (-> aabb types/max-p .x) grid-size)
-                      (grid-snap-up   (-> aabb types/max-p .y) grid-size)
-                      (grid-snap-up   (-> aabb types/max-p .z) grid-size))))
-
-(g/defnk update-grids
-  [camera]
-  (let [frustum-planes   (c/viewproj-frustum-planes camera)
-        far-z-plane      (nth frustum-planes 5)
-        normal           (as-unit-vector far-z-plane)
-        aabb             (frustum-projection-aabb frustum-planes)
-        extent           (geom/as-array (geom/aabb-extent aabb))
-        perp-axis        2
-        _                (aset-double extent perp-axis Double/POSITIVE_INFINITY)
-        smallest-extent  (reduce min extent)
+  [^AABB aabb size]
+  (types/->AABB (Point3d. (grid-snap-down (-> aabb types/min-p .x) (:x size))
+                          (grid-snap-down (-> aabb types/min-p .y) (:y size))
+                          (grid-snap-down (-> aabb types/min-p .z) (:z size)))
+                (Point3d. (grid-snap-up (-> aabb types/max-p .x) (:x size))
+                          (grid-snap-up (-> aabb types/max-p .y) (:y size))
+                          (grid-snap-up (-> aabb types/max-p .z) (:z size)))))
+
+(g/defnk produce-grids
+  [camera merged-options]
+  (let [{:keys [size active-plane auto-scale]} merged-options
+        frustum-planes (c/viewproj-frustum-planes camera)
+        plane (.indexOf axes active-plane)
+        aabb (frustum-projection-aabb frustum-planes plane)
+        extent (geom/as-array (geom/aabb-extent aabb))
+        _ (aset-double extent plane Double/POSITIVE_INFINITY)
+        smallest-extent (reduce min extent)
         first-grid-ratio (grid-ratio smallest-extent)
-        grid-size-small  (small-grid-size smallest-extent)
-        grid-size-large  (large-grid-size smallest-extent)]
-    {:ratios [first-grid-ratio                        (- 1.0 first-grid-ratio)]
-     :sizes  [grid-size-small                         grid-size-large]
-     :aabbs  [(snap-out-to-grid aabb grid-size-small) (snap-out-to-grid aabb grid-size-large)]}))
+        grid-size-small (grid-size size auto-scale smallest-extent small-grid-axis-size)
+        grid-size-large (when auto-scale (grid-size size auto-scale smallest-extent large-grid-axis-size))]
+    {:ratios [first-grid-ratio (- 1.0 ^double first-grid-ratio)]
+     :sizes [grid-size-small grid-size-large]
+     :aabbs (cond-> [(snap-out-to-grid aabb grid-size-small)]
+              grid-size-large
+              (conj (snap-out-to-grid aabb grid-size-large)))
+     :plane plane}))
+
+(g/defnk produce-merged-options
+  [prefs camera options]
+  (cond-> (if prefs (prefs/get prefs grid-prefs-path) {})
+    :always
+    (assoc :auto-scale true)
```

Comment:
The auto scale is not part of the grid prefs (it doesn't need to be), so this provides the default.